File: core/src/main/java/org/selyu/smp/core/menu/Menus.java
package org.selyu.smp.core.menu;

import fr.minuskube.inv.ClickableItem;
import org.bukkit.Material;
import org.bukkit.inventory.ItemStack;
import org.bukkit.inventory.meta.ItemMeta;
import org.jetbrains.annotations.NotNull;

import static org.selyu.smp.core.util.BukkitUtil.ensureMeta;
import static org.selyu.smp.core.util.MessageUtil.color;

public final class Menus {
    public static final ClickableItem PLACEHOLDER_ITEM = ClickableItem.empty(buildMenuItem(Material.BLACK_STAINED_GLASS_PANE, "&0"));
    public static final ItemStack NEXT_PAGE_ITEM_STACK = buildMenuItem(Material.ARROW, "&a>>");
    public static final ItemStack PREV_PAGE_ITEM_STACK = buildMenuItem(Material.ARROW, "&b<<");
    public static final ItemStack BACK_ITEM_STACK = buildMenuItem(Material.BARRIER, "&cGo back!");

    private Menus() {
    }

    @NotNull
    private static ItemStack buildMenuItem(Material material, String displayName) {
        ItemStack itemStack = new ItemStack(material);
        ItemMeta itemMeta = ensureMeta(itemStack);
        itemMeta.setDisplayName(color(displayName));

        itemStack.setItemMeta(itemMeta);
        return itemStack;
    }
}

File: core/src/main/kotlin/org/selyu/smp/core/menu/menus.kt
package org.selyu.smp.core.menu

import fr.minuskube.inv.ClickableItem
import org.bukkit.Material
import org.bukkit.inventory.ItemStack
import org.selyu.smp.core.util.color
import org.selyu.smp.core.util.ensureMeta

val placeholderItem: ClickableItem = ClickableItem.empty(ItemStack(Material.GRAY_STAINED_GLASS_PANE).also {
    it.ensureMeta()

    val meta = it.itemMeta!!
    meta.setDisplayName("&0".color)
    it.itemMeta = meta
})

val nextPageItemStack: ItemStack = ItemStack(Material.ARROW).also {
    it.ensureMeta()

    val meta = it.itemMeta!!
    meta.setDisplayName("&a>>".color)
    it.itemMeta = meta
}

val prevPageItemStack: ItemStack = ItemStack(Material.ARROW).also {
    it.ensureMeta()

    val meta = it.itemMeta!!
    meta.setDisplayName("&b<<".color)
    it.itemMeta = meta
}

val backItemStack: ItemStack = ItemStack(Material.BARRIER).also {
    it.ensureMeta()

    val meta = it.itemMeta!!
    meta.setDisplayName("&cGo back!".color)
    it.itemMeta = meta
}